src/devices/tuya.ts

```diff
@@ -8734,6 +8734,45 @@ const definitions: Definition[] = [
             ],
         },
     },
+    {
+        fingerprint: tuya.fingerprint('TS0601', ['_TZE204_rzrrjkz2']),
+        model: 'TS0601',
+        vendor: 'Tuya',
+        fromZigbee: [tuya.fz.datapoints],
+        toZigbee: [tuya.tz.datapoints],
+        description: 'Smart Sprinkler Timer',
+        onEvent: tuya.onEventSetTime,
+        configure: tuya.configureMagicPacket,
+        exposes: [
+            e.switch(),
```

Comment:
I'm missing a lot of exposes here, could you add them?

Reply:
Hi Koenkk, 

Thank you for the fast respond, really appreciated,

I must say that I'm really new to GitHub and actually have a little knowledge on how to make exposes works. 
The data point IDs are extracted from the device in Tuya Cloud Project, as I'm following from this instruction
https://www.zigbee2mqtt.io/advanced/support-new-devices/02_support_new_tuya_devices.html#_3-understanding-tuya-datapoints

I tried to put a lot of exposes before but it seems to be rejected, that's why I have to remove them.

Could you please provide me any instructions on how to make exposes works?

I can see these information on Device Debugging on Tuya Cloud Project, Not sure if it should help us on developing exposes?

<html>
<body>
<!--StartFragment-->
Code | Type | Values
-- | -- | --
switch | Boolean | "{true,false}"
status | Enum | {   "range": [     "off",     "auto",     "disabled"   ] }
countdown | Integer | {   "unit": "min",   "min": 1,   "max": 240,   "scale": 0,   "step": 1 }
countdown_left | Integer | {   "unit": "min",   "min": 1,   "max": 240,   "scale": 0,   "step": 1 }
water_current | Integer | {   "unit": "gal/min",   "min": 0,   "max": 1000000000,   "scale": 3,   "step": 1 }
battery_percentage | Integer | {   "unit": "%",   "min": 0,   "max": 100,   "scale": 0,   "step": 1 }
water_total | Integer | {   "unit": "gal",   "min": 0,   "max": 999999999,   "scale": 3,   "step": 1 }
weather_delay | Enum | {   "range": [     "24h",     "48h",     "72h",     "cancel"   ] }
normal_timer | String | {   "maxlen": 255 }
weather_switch | Boolean | "{true,false}"
switch_enabled | Boolean | "{true,false}"

<!--EndFragment-->
</body>
</html>

Reply:
> I tried to put a lot of exposes before but it seems to be rejected, that's why I have to remove them.

What do you mean with "rejected" here? 

For examples, please look at e.g. this:

expose
https://github.com/Koenkk/zigbee-herdsman-converters/blob/4812c1bc77683beb786644de86ac8c642d895a50/src/devices/tuya.ts#L1104

datapoint:
https://github.com/Koenkk/zigbee-herdsman-converters/blob/4812c1bc77683beb786644de86ac8c642d895a50/src/devices/tuya.ts#L1127